www_src/components/color-spectrum/color-spectrum.jsx

```diff
@@ -1,20 +1,116 @@
 var React = require('react');
+var Slider = require('../../components/range/range.jsx');
+var Color = require('color');
 
-// Doesn't do much yet, but renders the hue and saturation spectrum
+var HSB = React.createClass({
+  getBackgroundColor: function () {
+    var color = this.props.color.clone();
+    color.hsv(color.hue(), 100, 100);
+    return color.rgbaString();
+  },
+  getSBPosition: function () {
+    var hsv = this.props.color.hsv();
+    return {
+      top: `${100 - hsv.v}%`,
+      left: `${hsv.s}%`
+    };
+  },
+  changeHue: function (e) {
+    var color = this.props.color;
+    color.hue(e.target.value);
+    this.props.onChange(color);
+  },
+  changeSB: function (e) {
+    var color = this.props.color;
 
-var ColorSpectrum = React.createClass({
+    var x = e.touches[0].clientX - this.bounding.left;
+    var y = e.touches[0].clientY - this.bounding.top;
+
+    color.hsv(color.hue(), x / this.bounding.width * 100, 100 - y / this.bounding.height * 100);
```

Comment:
not really, the multiplication after the division throws me off. I can't tell if you really mean `x / this.bounding.width * 100` or whether you meant `x / (this.bounding.width * 100)`, which is made apparent my moving the multiplication up front: `100 * x / this.bounding.width`. (Slightly better performance that way too, since you're not upscaling a lossy floating point, but simply dividing a safe integer)